src/components/MoneyRequestConfirmationList.tsx

```diff
@@ -705,21 +706,7 @@ function MoneyRequestConfirmationList({
                 setFormError('iou.error.invalidCategoryLength');
                 return;
             }
-
-            if (formError) {
-                return;
-            }
-
-            if (iouType === CONST.IOU.TYPE.PAY) {
-                if (!paymentMethod) {
-                    return;
-                }
-
-                setDidConfirm(true);
-
-                Log.info(`[IOU] Sending money via: ${paymentMethod}`);
-                onSendMoney?.(paymentMethod);
-            } else {
+            if (iouType !== CONST.IOU.TYPE.PAY) {
```

Comment:
Just to confirm that we are not doing this by mistake. It is intentional that we don't check `formErrors` when `iouType !== CONST.IOU.TYPE.PAY`, right?

We don't need

```
            if (formError) {
                return;
            }
``` 
inside this `if`. Correct?

Reply:
> Just to confirm that we are not doing this by mistake. It is intentional that we don't check formErrors when iouType !== CONST.IOU.TYPE.PAY, right?

@aldo-expensify  The block after check `formErrors` will do nothing if `iouType !== CONST.IOU.TYPE.PAY` so we don't need to check this for the case `iouType !== CONST.IOU.TYPE.PAY`.

```
if (formError) {
    return;
}
```

Reply:
> The block after check `formErrors` will do nothing

Which block is that? I'm missing what you are referring to. I see that we do stuff in `iouType !== CONST.IOU.TYPE.PAY` even if there are `formErrors`, is that fine? This code will run:

<img width="1226" alt="image" src="https://github.com/Expensify/App/assets/87341702/243382fb-80f8-4705-b68a-971f64790cb8">


right?

Reply:
> Which block is that

@aldo-expensify  That is here. With the change of this PR we only do this after the check `formError`

```
if (iouType === CONST.IOU.TYPE.PAY) {
    if (!paymentMethod) {
        return;
    }

    setDidConfirm(true);

    Log.info(`[IOU] Sending money via: ${paymentMethod}`);
    onSendMoney?.(paymentMethod);
}
```

> I see that we do stuff in iouType !== CONST.IOU.TYPE.PAY even if there are formErrors, is that fine

That is the aim of this PR to fix the issue, we should move the logic to setting the error before the check `formError`

Reply:
oh, I see what you are saying now, thanks for explaining!